library/patient_self_checkin/step5.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+use OpenEMR\Core\Header;
+
+$ignoreAuth = true;
+require_once "../../interface/globals.php";
+if (!$GLOBALS['self_checkin_enable'] == 1) {
+    die("This feature has not been enabled");
+}
+?>
+
+<link rel="stylesheet" type="text/css" href="/interface/themes/selfCheckIn.css">
+
+<!DOCTYPE html>
+<html>
+<head>
+
+    <title><?php echo xlt("Check In"); ?></title>
+<?php
+
+Header::setupHeader();
+
+?>
+</head>
+<body class="step5">
+
+<p>
+    <img style="text-align: center"
+         width="500px" src='<?php echo $GLOBALS['images_static_relative']; ?>/logo-full-con.png'/>
+</p>
+
+<?php
+
+$month = htmlspecialchars($_GET['month']);
+$birthday = htmlspecialchars($_GET['date']);
+$surname = htmlspecialchars($_GET['surname']);
+
+// date_default_timezone_set();
+
+$timeRN = date("Y-m-d H:i:s");
+$date = date("y-m-d");
+echo "<h1> Time: " . $timeRN . "</h1><br>";
+
+require 'mySQL_connector.php';
+
+// Get the patient IDs of everyone with an appointment today. Enter them into an array which we can call upon later.
+
+$todaysPIDsQuery = "SELECT pc_pid from `openemr_postcalendar_events` WHERE `pc_eventDate` = '" . $date ."'";
+
+$todaysPIDs_result = mysqli_query($dbc, $todaysPIDsQuery); // This line executes the MySQL query that you typed above
+
+$PIDArray = array(); // make a new array to hold all your data
+
+$index = 0;
+while ($row = mysqli_fetch_array($todaysPIDs_result)) { // loop to store the data in an associative array.
+     $PIDArray[$index] = $row;
+     $index++;
+}
+
+// Create an array which we can pass to the subsequent MySQL query.
+
+$PIDArrayValues = array_map('array_pop', $PIDArray);
+$implodedArray = implode(',', $PIDArrayValues);
+
+// Get patient ID based on the data provided.
+// Criteria: a) has an appointment today i.e. their patient ID is one of the ones in the appointment list for today
+// b) Matches all three demographics asked earlier - dob, surname
+
+$PIDQuery = "SELECT pid
+FROM patient_data
+WHERE lname LIKE '" . $surname . "%' 
+AND dob LIKE '%-" . $month ."-" . $birthday . "'  
+AND id IN (".$implodedArray.")";
+
+$patient_id = mysqli_fetch_array(mysqli_query($dbc, $PIDQuery));
```

Comment:
binding and using openemr sql functions will change this to:
``````
$PIDQuery = "SELECT pid
FROM patient_data
WHERE lname LIKE ? 
AND dob LIKE ?  
AND id IN (?)";

$patient_id = sqlStatement($PIDQuery, array($surname . "%", "%-" . $month . "-" . $birthday, $implodedArray));

Reply:
Quick question here. Are you only meaning to collect one entry here. If so, then we can use sqlQuery() function which will return an array of the entry, thus would have !empty($sqlrtn['pid'])